secboot/secboot_hooks.go

```diff
@@ -174,9 +166,25 @@ func unlockVolumeUsingSealedKeyFDERevealKeyV2(sealedEncryptionKeyFile, sourceDev
 		return res, xerrors.Errorf(fmt, err)
 	}
 
+	model, err := opts.WhichModel()
+	if err != nil {
+		return res, fmt.Errorf("cannot retrieve which model to unlock for: %v", err)
+	}
+
 	// the output of fde-reveal-key is the unsealed key
 	options := activateVolOpts(opts.AllowRecoveryKey)
-	modChecker, err := sbActivateVolumeWithKeyData(mapperName, sourceDevice, keyData, options)
+	// FIXME: consider setting it in activateVolOpts if we keep
+	// this function separate from the tpm one for key data in
+	// files. Otherwise we should only set it if we provide key
+	// data generation 1.
```

Comment:
I'd like to actually remove the `Model` field in secboot given that future versions of snap-bootstrap are going to call `bootscope.SetModel`. We should probably obtain the model from there for existing keys rather than have a separate option.

Reply:
That would be great.